worker/src/backgroundMigrations/20250102_backfillBillingCycleAnchors.ts

```diff
@@ -0,0 +1,346 @@
+import { IBackgroundMigration } from "./IBackgroundMigration";
+import { logger } from "@langfuse/shared/src/server";
+import { parseArgs } from "node:util";
+import { prisma } from "@langfuse/shared/src/db";
+import { parseDbOrg } from "@langfuse/shared";
+import Stripe from "stripe";
+import { env } from "../env";
+
+const stripeClient = env.STRIPE_SECRET_KEY
+  ? new Stripe(env.STRIPE_SECRET_KEY)
+  : undefined;
+
+/**
+ * Background migration to backfill billing cycle anchors for organizations.
+ *
+ * Background:
+ * - Organizations created before the billing cycle anchor feature need to be backfilled
+ * - For hobby plan users: set anchor to organization createdAt
+ * - For paid plan users: fetch from Stripe subscription and use billing_cycle_anchor
+ *
+ * This migration is idempotent and can be safely re-run if interrupted.
+ */
+export default class BackfillBillingCycleAnchors
+  implements IBackgroundMigration
+{
+  private isAborted = false;
+
+  async validate(
+    // eslint-disable-next-line no-unused-vars
+    args: Record<string, unknown>,
+  ): Promise<{ valid: boolean; invalidReason: string | undefined }> {
+    // If not in cloud environment, validation passes (will skip in run())
+    if (!env.NEXT_PUBLIC_LANGFUSE_CLOUD_REGION) {
+      logger.info(
+        "[Background Migration] Not in cloud environment, migration will be skipped",
+      );
+      return { valid: true, invalidReason: undefined };
+    }
+
+    // In cloud environment: Stripe is required
+    if (!env.STRIPE_SECRET_KEY) {
+      return {
+        valid: false,
+        invalidReason:
+          "Migration requires Stripe integration in cloud environment (STRIPE_SECRET_KEY not set)",
+      };
+    }
+
+    return { valid: true, invalidReason: undefined };
+  }
```

Comment:
Theoretically, it would make sense to check that the new columns exist in the DB to avoid race-conditions between the prisma migrations and the background migrations. As we have control on cloud this should be acceptable if we deploy it correctly.